src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_1001_2000.cpp

```diff
@@ -492,21 +492,24 @@ GUI_TEST_CLASS_DEFINITION(test_1021_3) {
 
         // 2) Click "build dotplot" tooltip
         // 3) Click OK in opened dotplot dialog
-        GTUtilsDialog::waitForDialog(os, new DotPlotFiller(os, 120, 100, true));
+        GTUtilsDialog::add(os, new DotPlotFiller(os, 120, 100, true));
         GTWidget::click(os, GTWidget::findWidget(os, "build_dotplot_action_widget"));
+        GTThread::waitForMainThread();
 
         if (i == 0) {
             // GTUtilsMdi::click(os, GTGlobals::Minimize);
             GTGlobals::sleep();
         }
 
         // 4) Click on human_T1.fa project tree view item
-        GTUtilsDialog::waitForDialog(os, new MessageBoxDialogFiller(os, QMessageBox::No, "Save dot-plot data before closing?"));
+        GTUtilsDialog::add(os, new MessageBoxDialogFiller(os, QMessageBox::No, "Save dot-plot data before closing?"));
```

Comment:
nit: these changes are no-op: when you have only 1 waiter it does not matter if you are using add or waitForDialog

Reply:
I know